fe/src/main/java/org/apache/doris/transaction/GlobalTransactionMgr.java

```diff
@@ -851,493 +243,114 @@ public boolean isIntersectionNotEmpty(List<Long> sourceTableIdList, List<Long> t
      */
     public void removeExpiredAndTimeoutTxns() {
         long currentMillis = System.currentTimeMillis();
-
-        List<Long> timeoutTxns = Lists.newArrayList();
-        List<Long> expiredTxns = Lists.newArrayList();
-        readLock();
-        try {
-            for (TransactionState transactionState : idToTransactionState.values()) {
-                if (transactionState.isExpired(currentMillis)) {
-                    // remove the txn which labels are expired
-                    expiredTxns.add(transactionState.getTransactionId());
-                } else if (transactionState.isTimeout(currentMillis)) {
-                    // txn is running but timeout, abort it.
-                    timeoutTxns.add(transactionState.getTransactionId());
+        for (DatabaseTransactionMgr dbTransactionMgr : dbIdToDatabaseTransactionMgrs.values()) {
+            dbTransactionMgr.removeExpiredTxns();
+            List<Long> timeoutTxns = dbTransactionMgr.getTimeoutTxns(currentMillis);
+            // abort timeout txns
+            for (Long txnId : timeoutTxns) {
```

Comment:
This logic(Remove the timeout txn) can also be put into `DatabaseTransactionMgr`.
`dbTransactionMgr. removeExpiredAndTimeoutTxns();`